### DISCUSSION TITLE
queues in Pocketbase

### DISCUSSION THREAD

**Discussion by tigawanna** [pocketbase/pocketbase#4547] [Q&A] [answered]
has anyone implemented queues , like you have an endpoint to enqueue the messages and a listener of some sort that excecutes them when it's their turn

**Comment by whmeitzler** [comment, marked as answer]
Message Queues within a single daemon have limited functionality, it depends on what you're trying to accomplish. People often will use something like asynq to pass tasks to a place where they'll live durably, until distributed to a worker running as a separate process.

If you're looking to fan out events, I've had luck using Pocketbases hooks pattern with custom types. That gives you an in-process pub/sub model.

For queuing tasks, I've had luck making a collection to capture ... whatever the task is, and having one process pick up records from that collection to try to execute them.

Whatever you're setting out to do, really think about why you're looking for a message queue. There's lots of good reasons to use one... and there's a lot of ways to solve smaller problems by extending pocketbase with pure Go.

Good luck!

**Reply by joseferben** [reply]
> For queuing tasks, I've had luck making a collection to capture ... whatever the task is, and having one process pick up records from that collection to try to execute them.

I've implemented this as library: https://github.com/joseferben/pocketbase-queue/
The locking is a bit heavy, it would be cool to have some kind of compare-and-swap semantics in the CRUD API, maybe returning a flag whether a collection was updated or not.

**Reply by gokulgseven** [reply]
as [whmeitzler](https://github.com/whmeitzler) said 
create a collection to capture the requests , then use cron jobs to one by one execute the Queues. 
https://pocketbase.io/docs/js-jobs-scheduling/ documentation for cron jobs
